pkg/pgxconn/pgx_conn.go

```diff
@@ -109,15 +147,53 @@ func (p *connImpl) Close() {
 }
 
 func (p *connImpl) Exec(ctx context.Context, sql string, args ...interface{}) (pgconn.CommandTag, error) {
-	return p.Conn.Exec(ctx, sql, args...)
+	lset := getLabelSet("exec")
+	requestTotal.With(lset).Inc()
+	start := time.Now()
+	defer func() {
+		duration.With(lset).Observe(time.Since(start).Seconds())
+	}()
+	tag, err := p.Conn.Exec(ctx, sql, args...)
+	if err != nil {
+		errorsTotal.With(lset).Inc()
+	}
+	return tag, err
 }
 
 func (p *connImpl) Query(ctx context.Context, sql string, args ...interface{}) (PgxRows, error) {
-	return p.Conn.Query(ctx, sql, args...)
+	lset := getLabelSet("query")
+	requestTotal.With(lset).Inc()
+	start := time.Now()
+	defer func() {
+		duration.With(lset).Observe(time.Since(start).Seconds())
+	}()
+	rows, err := p.Conn.Query(ctx, sql, args...)
+	if err != nil {
+		errorsTotal.With(lset).Inc()
+	}
+	return rows, err
 }
 
 func (p *connImpl) QueryRow(ctx context.Context, sql string, args ...interface{}) pgx.Row {
-	return p.Conn.QueryRow(ctx, sql, args...)
+	lset := getLabelSet("query_row")
+	requestTotal.With(lset).Inc()
+	start := time.Now()
+	defer func() {
+		duration.With(lset).Observe(time.Since(start).Seconds())
+	}()
+	return &rowWrapper{p.Conn.QueryRow(ctx, sql, args...)}
+}
+
+type rowWrapper struct {
+	r pgx.Row
+}
+
+func (w *rowWrapper) Scan(dest ...interface{}) error {
+	err := w.r.Scan(dest...)
+	if err != nil {
```

Comment:
We should make sure that `err != ErrNoRows`. Based on `QueryRow` docs it returns error when there is no data.

Reply:
yeah, i just fixed this. There can be response as empty if the metric queried in promql, no such metric was ever ingested.

Reply:
Ah, so it is more of a client-side error (similar to 404 in HTTP). SGTM to drop it and not report in `errorsTotal` (at least for now).